### ISSUE TITLE
Closing first Query tab never warns of unsaved changes.

### ISSUE THREAD

**Issue by yphastos** [HeidiSQL/HeidiSQL#231]
## Steps to reproduce this issue

1. Write any content on the first Query tab (always labeled 'Query')
2. Close the tab by either double clicking it, or by closing HeidiSQL. Note that for some reason, that first Query tab doesn't have a close button, neither right clicking it shows the 'Close query tab' option (is grayed out). But it can be closed by double clicking or closing Heidi.
3. Despite the tab not being saved (and showing pending changes with a '\*' in the tab's title ('Query*'), the application doesn't warn that there is unsaved changes, and closes it regardless.

## Current behavior

The tab closes without any warning of unsaved content.

## Expected behavior

Attempting to close the tab should warn of unsaved changes and prompt if user wants to save content.

## Possible solution

Validate unsaved changes as any other tab,.

## Environment

* **HeidiSQL version:** 9.5.0.5273
  
* **Database system and version:** MySQL 5.1.73
  
* **Operating system:** MacOS using wine. Also happens in windows

**Comment by ansgarbecker** [comment]
You probably just disabled the prompt for it, in Preferences > SQL:

![grafik](https://user-images.githubusercontent.com/7986591/38894935-f5ead64c-428e-11e8-99ae-1c7bd0cb2e91.png)

**Comment by yphastos** [comment]
NO! the option IS active, and it DOES work for every other tab. 

But it does NOT work in THE FIRST query tab (it doesn't prompt and clears the content).

I mentioned that particularity in the problem description. it only happens with the first tab.

Please try to replicate it, just write anything on the first Query tab, and without saving the content, **double click** the tab itself. It will clear the tab contents without prompting or warning anything. The tab itself never closes, but the content does disappear.

Please reopen the issue since it is not solved.

**Comment by ansgarbecker** [comment]
Ah, I see now. I did reproduce that, but only by closing HeidiSQL, and then got this:
![grafik](https://user-images.githubusercontent.com/7986591/38912546-2b331f7a-42d6-11e8-8323-ab25c67a1be8.png)
You wrote
> Close the tab by either double clicking it, or by closing HeidiSQL.
I did not do that doubleclick again. And yes, it's not warning me when doubleclicking the tab. The reason for not asking on that special first tab was that it is not closable anyway (intentionally). So, after the content is gone, you can get it back via undo (<kbd>Ctrl+Z</kbd>).

**Comment by yphastos** [comment]
I see,
But shouldn't it warn anyway that the contents would be cleared (despite the tab not closing itself)? 

Even if we can get the content back with Ctrl+Z (which btw is quite unintuitive), i think it should always warn if the contents would be lost (or clear) for whichever tab (honoring the setting).

Or if the tab is meant to never close, it then never should clear the content by double clicking it (just as it doesn't have the 'Close query tab' item when right clicking, or not having the 'x' control to close it) That way, the only method to close the tab (and potentially lose content) would be when closing Heidi.

**Comment by ansgarbecker** [comment]
Oh I think that's a good idea - just not remove the content. That way the doubleclick feature is clearly only for closable tabs.

**Comment by albertdick** [comment]
Well, it's bad. I think that first tab should be "closed" in some way.

The impression I have is @yphastos accidentally clicked twice on the first tab.

I've used HeidiSQL for almost two years, almost every day, and I've never accidently closed the first tab.

Now, even if I close all used tabs, I'll see a warning (that I don't want to disable) because I can't "close" the first tab anymore.

**Comment by anselal** [comment]
Or it could just warn you and you can leave the double click feature as it is..

**Comment by anselal** [comment]
@ansgarbecker  The problem is still there in the latest build 9.5.0.5287

By pressing Ctrl+W the tab just clears without any warning for modified content.

**Comment by yphastos** [comment]
@anselal Wow, that's true!

I guess the behavior is that the tab attempts to close, but since it can't, it somehow doesn't warn anything but it still clears its content. (the same that happened when double-clicking it)

Maybe the tab should be allowed to close just as any other, triggering the same warnings and behaviors as the rest (having a close button, have the "close query tab" right click item, closing via double-click or Ctrl+W (as it is common in tons of programs), etc. 

I don't know the reason it is not meant to close, @ansgarbecker has mentioned it is intentional, but personally I see no problem in closing it (from an end user point of view), I don't know if there are technical reasons to not close it. 

But ultimately, there is always the "open a blank query tab" button in the tab bar, as well as the "open query tab - Ctrl+T" item in the File menu. So even if someone closes ALL tabs, you can always open new ones.

So I vote to treat the first tab exactly as all others, with the same behaviors and warnings, including closing it.

Greetings.

**Comment by ansgarbecker** [comment]
The first tab is also sticky to help newbies. And there are other reasons. Most of the time many users just need one tab. Having to create one is an additional click.

**Comment by yphastos** [comment]
IMO you could always create one initial empty one, as it is now. But allow to close it as any other tabs. 

If users close it, it would be on purpose, not by accident (because attempting to close it with unsaved data will warn them), and they would need to open a new one on their own, as expected.

I think it is more intuitive that way instead of accidentally pressing Ctrl+W and losing the unsaved content (without a warning), not being aware that Ctrl+Z could restore it.

**Comment by albertdick** [comment]
I think it easier to close a tab by accident in the traditional way (by clicking on the "x") than pressing <kbd>CTRL</kbd>+<kbd>W</kbd>.

**Comment by yphastos** [comment]
@albertdick indeed! but its the same point. I mean you can 'accidentally' attempt to close it, without saving changes. If it was already saved, its no problem because you could simply reopen it.

If the tab is 'closeable', it *will* warn of unsaved changes (by any method, 'x' corner, double-click, ctrl+w, and whatever other method for closing tabs I'm unaware of).

But if it is *not* closable, it simply *doesn't warn* anything and clears the content (and keeps the tab). 
THAT is the main problem.

Of course all the "close" methods could be disabled for the first tab, in this case Ctrl+W. Just as the 'x' and double click are already disabled.

But IMO its non ideal. It is not intuitive at all not being able to close every single tab. and even less that a "closing" event doesn't warn anything and just clears the content.

**Comment by ansgarbecker** [comment]
As just written in #318 - double click works in the newest nightly build again as previously ;)
With one exception: it asks for confirmation now, like when you close a tab with unsaved contents.

For modification details see my commit 42a03dfe83cee70d3646d5100b610599c81547c5

**Comment by anselal** [comment]
Those are great news. Thanks !!! Gonna try it out

**Comment by anselal** [comment]
Works like a charm !!!
